File: backend/src/main/java/ch/rodano/core/services/bll/configurator/ChartService.java
package ch.rodano.core.services.bll.configurator;

import java.util.List;
import java.util.UUID;

import ch.rodano.api.config.ChartModelDTO;

public interface ChartService {

	List<ChartModelDTO> getCharts(UUID projectId, String view);

	default List<ChartModelDTO> getCharts(final UUID projectId) {
		return getCharts(projectId, "summary");
	}

	ChartModelDTO getChart(UUID projectId, UUID chartId);

	ChartModelDTO createChart(UUID projectId, ChartModelDTO chart);

	ChartModelDTO updateChart(UUID projectId, UUID chartId, ChartModelDTO chart);

	void deleteChart(UUID projectId, UUID chartId);
}

File: frontends/main/src/app/configurator/services/api/chart.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {ChartModel} from '@core/model/chart-model';

@Injectable({
	providedIn: 'root'
})
export class ChartService {
	constructor(private http: HttpClient) {}

	getCharts(projectId: string): Observable<ChartModel[]> {
		return this.http.get<ChartModel[]>(`/api/superuser/configurator/projects/${projectId}/config/charts`, {
			params: {view: 'summary'}
		});
	}

	getChartsFull(projectId: string): Observable<ChartModel[]> {
		return this.http.get<ChartModel[]>(`/api/superuser/configurator/projects/${projectId}/config/charts`, {
			params: {view: 'full'}
		});
	}

	getChart(projectId: string, chartId: string): Observable<ChartModel> {
		return this.http.get<ChartModel>(`/api/superuser/configurator/projects/${projectId}/config/charts/${chartId}`);
	}

	createChart(projectId: string, chart: ChartModel): Observable<ChartModel> {
		return this.http.post<ChartModel>(`/api/superuser/configurator/projects/${projectId}/config/charts`, chart);
	}

	updateChart(projectId: string, chartId: string, chart: ChartModel): Observable<ChartModel> {
		return this.http.put<ChartModel>(`/api/superuser/configurator/projects/${projectId}/config/charts/${chartId}`, chart);
	}

	deleteChart(projectId: string, chartId: string): Observable<void> {
		return this.http.delete<void>(`/api/superuser/configurator/projects/${projectId}/config/charts/${chartId}`);
	}
}
